src/constructs/sns/sns-topic.ts

```diff
@@ -1,10 +1,26 @@
-import { Topic } from "@aws-cdk/aws-sns";
+import { AccountPrincipal, OrganizationPrincipal } from "@aws-cdk/aws-iam";
 import type { TopicProps } from "@aws-cdk/aws-sns";
+import { Topic } from "@aws-cdk/aws-sns";
+import { RegexPattern } from "../../constants";
 import { GuStatefulMigratableConstruct } from "../../utils/mixin";
 import type { GuStack } from "../core";
 import type { GuMigratingResource } from "../core/migrating";
 
-interface GuSnsTopicProps extends TopicProps, GuMigratingResource {}
+interface GuSnsTopicProps extends TopicProps, GuMigratingResource {
+  /**
+   * List of AWS account ids to grant `SNS:Publish` permission to.
+   *
+   * Note: if you need to add conditions to the Topic Policy, consider calling `this.grantPublish` directly.
+   */
+  accountsAllowedToPublish?: string[];
```

Comment:
I wonder if it'd be better to surface this at the [pattern level](https://github.com/guardian/cdk/blob/794ff425b8e540807833b8f514f460e8d15d1cdf/src/patterns/sns-lambda.ts#L69-L72) instead/as well?

Reply:
Agreed. I think ideally a pattern's props should extend all the props of it's child construct's props to keep things a bit DRY. This would be quite the refactor so I'd be tempted to do this in a separate PR. WDYT?

Reply:
> I think ideally a pattern's props should extend all the props of it's child construct's props to keep things a bit DRY.

I'm not sure about surfacing _all_ props in all cases - that could produce some pretty overwhelming interfaces/documentation (I'm thinking about a complex pattern like `GuEc2App` which has many children here). Happy to think about how to apply this to the pattern in a separate PR though!